apps/class-solid/src/lib/store.ts

```diff
@@ -238,19 +210,23 @@ export function swapPermutationAndReferenceConfiguration(
   permutationIndex: number,
 ) {
   const exp = findExperiment(experimentIndex);
-  const refConfig = structuredClone(exp.reference.config);
-  const perm = exp.permutations[permutationIndex];
-  const permConfig = structuredClone(perm.config);
+  const refConfig = structuredClone(exp.config.reference);
+  const perm = exp.config.permutations[permutationIndex];
+  const permConfig = structuredClone(perm);
```

Comment:
The store is getting quite big for my complexity comfort level. Would it make sense to split it into dedicated modules for experiments, permutations, and analysis?